lib/timeline-window.js

```diff
@@ -254,7 +254,12 @@ TimelineWindow.prototype.paginate = function(direction, size, makeRequest,
 
     debuglog("TimelineWindow: starting request");
     var self = this;
-    var prom = this._client.paginateEventTimeline(tl.timeline, {
+
+    var paginateTimeline = tl.timeline.getRoomId() ?
```

Comment:
FTR, I'm sad that this has to happen here. I feel like this decision should be made by `MatrixClient.paginateEventTimeline`: it would clean up the API for `MatrixClient`